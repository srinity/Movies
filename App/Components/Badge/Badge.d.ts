import { Component } from 'react'
import { TextProps, ViewStyle, TextStyle, StyleProp } from 'react-native'

export interface BadgeProps extends Omit<TextProps, 'style'> {
  /**
   * The name that will appear in the badge
   *
   */
  name: string

  /**
   * optional custom styles for the container
   *
   */
  style: StyleProp<ViewStyle>

  /**
   * optional custom styles for the name text
   *
   */
  textStyle: StyleProp<TextStyle>
}

export default class Badge extends Component<BadgeProps> {}
