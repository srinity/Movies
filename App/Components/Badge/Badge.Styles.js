import { StyleSheet } from 'react-native'

import { Colors, Fonts } from '../../Theme'

export default StyleSheet.create({
  containerStyle: {
    backgroundColor: Colors.notAvailableColor,
    borderRadius: 12.5,
    marginRight: 5,
    paddingHorizontal: 10,
    paddingVertical: 2,
    marginTop: 7
  },
  textStyle: {
    color: Colors.blackColor,
    fontSize: Fonts.h7
  }
})
