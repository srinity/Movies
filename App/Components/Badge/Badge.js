import React from 'react'
import { View, Text, ViewPropTypes } from 'react-native'
import PropTypes from 'prop-types'

import styles from './Badge.Styles'

function Badge({ name, style, textStyle, ...props }) {
  return (
    <View style={[styles.containerStyle, style]}>
      <Text style={[styles.textStyle, textStyle]} {...props}>
        {name}
      </Text>
    </View>
  )
}

Badge.propTypes = {
  name: PropTypes.string,
  style: ViewPropTypes.style,
  textStyle: Text.propTypes.style
}

export default Badge
