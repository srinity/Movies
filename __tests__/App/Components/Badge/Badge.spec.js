import React from 'react'
import { Text, View } from 'react-native'
import renderer from 'react-test-renderer'

import Badge from '../../../../App/Components/Badge/Badge'

describe('Badge Component', () => {
  it('should render correctly', () => {
    const badge = renderer.create(<Badge name="Action" />)
    expect(badge).toMatchSnapshot()
  })

  it('should be able to customize textStyle correctly', () => {
    const textStyle = {
      color: '#e23478'
    }

    const badge = renderer.create(<Badge name="Action" textStyle={textStyle} />)
    expect(badge).toMatchSnapshot()

    expect(badge.root.findByType(Text).props.style[1]).toBe(textStyle)
  })

  it('should be able to customize style correctly', () => {
    const style = {
      backgroundColor: '#e23478'
    }

    const badge = renderer.create(<Badge name="Action" style={style} />)
    expect(badge).toMatchSnapshot()

    expect(badge.root.findByType(View).props.style[1]).toBe(style)
  })
})
