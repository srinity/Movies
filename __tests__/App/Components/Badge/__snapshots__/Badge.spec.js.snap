// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component should be able to customize style correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dddddd",
        "borderRadius": 12.5,
        "marginRight": 5,
        "marginTop": 7,
        "paddingHorizontal": 10,
        "paddingVertical": 2,
      },
      Object {
        "backgroundColor": "#e23478",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 10,
        },
        undefined,
      ]
    }
  >
    Action
  </Text>
</View>
`;

exports[`Badge Component should be able to customize textStyle correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dddddd",
        "borderRadius": 12.5,
        "marginRight": 5,
        "marginTop": 7,
        "paddingHorizontal": 10,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 10,
        },
        Object {
          "color": "#e23478",
        },
      ]
    }
  >
    Action
  </Text>
</View>
`;

exports[`Badge Component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dddddd",
        "borderRadius": 12.5,
        "marginRight": 5,
        "marginTop": 7,
        "paddingHorizontal": 10,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 10,
        },
        undefined,
      ]
    }
  >
    Action
  </Text>
</View>
`;
